File: frontend/src/Types/Recommendation/createRecommendationDto.ts
import { CriteriaEstimateDto } from '../Criteria/criteriaEstimateDto.ts';

export type CreateRecommendationDto = {
  businessId: string;
  criteriaEstimates: CriteriaEstimateDto[];
  pluses: string[];
  minuses: string[];
  recommendations: string[];
}

File: backend/ReadyBusinesses.Common/Dto/Recommendation/CreateRecommendationDto.cs
using ReadyBusinesses.Common.Dto.Criteria;

namespace ReadyBusinesses.Common.Dto.Recommendation;

public class CreateRecommendationDto
{
    public Guid BusinessId { get; set; }

    public IEnumerable<CriteriaEstimateDto> CriteriaEstimates { get; set; } = [];
    
    public bool ByAi { get; set; }

    public string[] Pluses { get; set; } = [];

    public string[] Minuses { get; set; } = [];

    public string[] Recommendations { get; set; } = [];
}